packages/core/src/tracing/backend_mappers/mapper.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * (c) Copyright IBM Corp. 2024
+ */
+
+'use strict';
+
+/**
+ * @typedef {Object<string, string>} FieldMapping
+ * @typedef {Object<string, FieldMapping>} FieldMappings
+ */
+
+/**
+ * Field mappings for different span types.
+ *
+ * This FieldMappings defines how internal span fields are mapped to backend fields
+ * for various span types (e.g., dynamodb, redis).
+ *
+ * As new span types needs to add, simply add their respective mappings
+ * following the same format (e.g., 'internal-field': 'backend-field').
+ *
+ * @type {FieldMappings}
+ */
+const fieldMappings = {
+  dynamodb: {
+    /// Maps internal field 'operation' to backend field 'op'
+    operation: 'op'
+  },
+  redis: {
+    operation: 'command'
+  }
+};
+
+/**
+ * Transforms span data fields to match the backend format.
+ *
+ * @param {import('../../core').InstanaBaseSpan} span
+ * @returns {import('../../core').InstanaBaseSpan} The transformed span.
+ */
+module.exports.transform = span => {
+  const spanName = span.n;
+  const mappings = fieldMappings[spanName];
+  if (!mappings || !span.data[spanName]) return span;
+
+  Object.keys(span.data[spanName]).forEach(internalField => {
```

Comment:
In general: It could be that a span contains multiple data keys. But I have never seen a case where two instrumentation keys were present. I think only with the SDK 🤔 
But I would ignore that for now I guess.